utils.js

```diff
@@ -37,10 +37,9 @@ const resolveObject = (object, context) => {
           throw Error(`the referenced substring is not a string`)
         }
       }
+      this.update(newValue)
     }
-    accum = assocPath(this.path, newValue, accum)
-    return accum
-  }, {})
+  })
 
   return resolvedObject
```

Comment:
interesting. If we're not reducing, how is the returned `resolvedObject` is correct? does `traverse.forEach` return the updated version?

Reply:
Yes, it does!

https://github.com/substack/js-traverse/blob/master/index.js#L6